### PR TITLE
dev/core#139 Fail more gracefully on contribution detail report with invalid combination

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Currently it is not valid to view 'Contributions only' with soft credit filters in the Contribution Detail report. This fix changes a hard error to a soft error when this is attempted

Before
----------------------------------------
Fatal error when combining 'Contributions only' with a filter on 'soft_credit_type_id'

After
----------------------------------------
Filter ignored with a message - no hard error

Technical Details
----------------------------------------
For better or worse this report munges soft credits & contributions in a very weird way.

There is already code to prevent accessing the soft credit criteria when doing 'Contributions only' search - but it was not working in this case. This PR fixes that & adds a message.

In general this report needs to be made testable before any notable changes are made. Moving some code to beginPostProcessCommon is a small step towards that.

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@jitendrapurohit are you able to review this?

File: CRM/Report/Form/Contribute/Detail.php [hunk 1]
```diff
@@ -577,6 +563,33 @@ public function postProcess() {
...
+    // when 'Contribution or Soft Credit?' column is not selected
+    if (empty($this->_params['fields']['contribution_or_soft'])) {
+      $this->_params['fields']['contribution_or_soft'] = 1;
+      $this->noDisplayContributionOrSoftColumn = TRUE;
+    }
+
+    if (CRM_Utils_Array::value('contribution_or_soft_value', $this->_params) ==
+      'contributions_only' &&
+      !empty($this->_params['fields']['soft_credit_type_id'])
+      || !empty($this->_params['soft_credit_type_id_value'])
```
Comment (jitendrapurohit) [inline CRM/Report/Form/Contribute/Detail.php]: Seems the `||` condition need to be inside brackets? Currently, the status message is shown if the soft credit type is selected in the filter with `Contribution OR Soft Credit?` set to any value. Works fine with below if statement -

    if (CRM_Utils_Array::value('contribution_or_soft_value', $this->_params) ==
      'contributions_only' &&
      (!empty($this->_params['fields']['soft_credit_type_id'])
      || !empty($this->_params['soft_credit_type_id_value']))

Comment (eileenmcnaughton) [reply CRM/Report/Form/Contribute/Detail.php]: Thanks @jitendrapurohit  - I just pushed a fix

**Review by jitendrapurohit — APPROVED** [review]
Tested contribution detail report with the patch, works as expected 👍

**Comment by yashodha** [comment]
@eileenmcnaughton 

I still get an error. My criteria is _Soft Credits Only_ and in columns I have selected all the soft credit fields
Soft Credit For, Soft Credits, Soft Credit Type

**Comment by eileenmcnaughton** [comment]
@yashodha do you want to submit a patch that extends this to suppress those fields too when it is 'Contributions only'

**Comment by eileenmcnaughton** [comment]
@yashodha Actually - we could probably use isTableSelected() rather than identifying fields?

**Pull request merged**